modules/infinite-scroll/infinity.php

```diff
@@ -1560,6 +1569,10 @@ public static function archive_supports_infinity() {
 	 * @return string or null
 	 */
 	function footer() {
+		if ( Jetpack_AMP_Support::is_amp_request() ) {
```

Comment:
We should check if class_exists here as per previous instruction on some of the other issues we've worked on